src/auth/googleauth.ts

```diff
@@ -286,6 +302,42 @@ export class GoogleAuth<T extends AuthClient = JSONClient> {
     return this._findProjectIdPromise;
   }
 
+  async #getUniverseFromMetadataServer() {
+    if (!(await this._checkIsGCE())) return;
+
+    let universeDomain: string;
+
+    try {
+      universeDomain = await gcpMetadata.universe('universe_domain');
+      universeDomain ||= DEFAULT_UNIVERSE;
+    } catch (e) {
+      if (e instanceof GaxiosError && e.status === 404) {
+        universeDomain = DEFAULT_UNIVERSE;
+      } else {
+        throw e;
+      }
+    }
+
+    return universeDomain;
+  }
+
+  /**
+   * Retrieves, caches, and returns the universe domain in the following order
+   * of precedence:
+   * - The universe domain in {@link GoogleAuth.clientOptions}
+   * - {@link gcpMetadata.universe}
+   *
+   * @returns The universe domain
+   */
+  async getUniverseDomain(): Promise<string> {
+    this.#universeDomain ??= originalOrCamelOptions(this.clientOptions).get(
+      'universe_domain'
+    );
+    this.#universeDomain ??= await this.#getUniverseFromMetadataServer();
```

Comment:
Would we not want to override value retrieved from the metadata server if someone explicitly sets universe domain via the options object? If not, I think that should be documented

Reply:
The documentation for this method mentions this precedence